source/_docs/mu-plugin.md

```diff
@@ -0,0 +1,256 @@
+ *
+ * For example, to exclude pages in the /news/ and /about/ path from cache, set:
+ *
+ *   $regex_path_patterns = array(
+ *     '#^/news/?#',
+ *     '#^/about/?#',
+ *   );
+ */
+
+
+// Loop through the patterns.
+foreach ($regex_path_patterns as $regex_path_pattern) {
+  if (preg_match($regex_path_pattern, $_SERVER['REQUEST_URI'])) {
+    add_action( 'send_headers', 'add_header_nocache', 15 );
+
+    // No need to continue the loop once there's a match.
+    break;
+  }
+}
+function add_header_nocache() {
+      header( 'Cache-Control: no-cache, must-revalidate, max-age=0' );
+}
+```
+
+### WP REST API (`wp-json`) Endpoints Cache
+
+For WP REST API endpoints, we can use the `rest_post_dispatch` filter and create a specific function to apply specific headers for each path or endpoint.
+
+```php
+/* For WP REST API specific paths, we use a different approach by using the rest_post_dispatch filter */
+
+// wp-json paths or any custom endpoints
+
+$regex_json_path_patterns = array(
+    '#^/wp-json/wp/v2/users?#',
+    '#^/wp-json/?#'
+    );
+  
+foreach ($regex_json_path_patterns as $regex_json_path_pattern) {
+  if (preg_match($regex_json_path_pattern, $_SERVER['REQUEST_URI'])) {
+    // re-use the rest_post_dispatch filter in the Pantheon page cache plugin  
+    add_filter( 'rest_post_dispatch', 'filter_rest_post_dispatch_send_cache_control', 12, 2 );
+    // Re-define the send_header value with any custom Cache-Control header
+    function filter_rest_post_dispatch_send_cache_control( $response, $server ) {
+      $server->send_header( 'Cache-Control', 'no-cache, must-revalidate, max-age=0' );
+      return $response;
+    }
+    break;
+  }
+}
+```
+
+### Exclude Plugins for Redis cache
+
+Referenced from [wp-redis](https://github.com/pantheon-systems/wp-redis#how-do-i-disable-the-persistent-object-cache-for-a-bad-actor){.external} on GitHub:
+
+A page load with 2,000 Redis calls can be 2 full seconds of object cache transactions. If a plugin you're using is erroneously creating a huge number of cache keys, you might be able to mitigate the problem by disabling cache persistency for the plugin's group:
+
+```php
+wp_cache_add_non_persistent_groups( array( 'bad-actor' ) );
+```
+
+This declaration means use of `wp_cache_set( 'foo', 'bar', 'bad-actor' );` and `wp_cache_get( 'foo', 'bad-actor' );` will not use Redis, and instead fall back to WordPress' default runtime object cache.
+
+### Setting Custom Cookies
+
+Setting custom cookies can also be done from an mu-plugin like, like the example below. Find more [cookie manipulation examples here](https://pantheon.io/docs/cookies/).
+
+```php
+if ( isset( $_COOKIE['STYXKEY_gorp'] ) ) {
+
+  $foo = $_COOKIE['STYXKEY_gorp'];
+  // Generate varied content based on cookie value
+  // Do NOT set cookies here; Set-Cookie headers do not allow the response to be cached
+  if ($foo == 'ca') {
+    str_replace('football', 'hockey');
+  }
+
+}
+else{
```

Comment:
Should probably have a ` ` (space) after the else statement -- depending on your coding standards you also might want to not have else on a new line?

Reply:
Spaced.